### ISSUE TITLE
CommonJS / component.io

### ISSUE THREAD

**Issue by ronkorving** [brix/crypto-js#5]
Hi, have you considered making this library `require`able?

**Comment by ghost** [comment]
This library is `require`able, just have a look into the zip archive in the build directory.

**Comment by ronkorving** [comment]
Sorry, let me rephrase. Have you considered adding a component.json file? I haven't started using this library yet, so am not sure exactly what that would look like.

**Comment by ghost** [comment]
I haven't considered yet, but it's a nice idea. Actually the mentioned zip archive is ready to use for RequireJS, so it should be not too tricky to make it available for component.io.

Do you have some hints? Do you already work with component.io?

**Comment by ghost** [comment]
Ok, I think I misunderstood component.io, it's not using RequireJS (async) style.

**Comment by ronkorving** [comment]
Thanks for your response. Yeah, I use component a lot. It has a synchronous `require` API, just like Node.js and browserify. And `component build` turns your components into a downloadable build.

A good place to start:

https://github.com/component/component/blob/master/Readme.md
https://github.com/component/component/wiki

**Comment by ghost** [comment]
This library supports:
CommonJS: `var CryptoJS = require('crypto-js');`
AMD: `require(['crypto-js'], function (CryptoJS) { ... });`

But it provides a modularized port of the crypto-js project, so there is no single main build.

For example:
`require('crypo-js');` loads the whole library (47 files)
`require('crypt-js/aes');` loads only the required modules for AES encryption (7 files)

I tried to provide a `component.json` for this repo, but I didn't get it working. Probably I've to provide a single main build of the library, but this would destroy the concept of the modularization.

**Comment by Neustradamus** [comment]
@ronkorving: I have done an important announcement here:
- https://github.com/brix/crypto-js/issues/536

The solution is to replace by crypto-es:
- https://github.com/entronad/crypto-es
- https://github.com/entronad/crypto-es/tags
- https://github.com/entronad/crypto-es/releases
